polkadot/node/core/approval-voting-parallel/src/lib.rs

```diff
@@ -0,0 +1,950 @@
+			.map_err(|err| SubsystemError::QueueError(err.into_send_error()))
+	}
+
+	fn meter(&self) -> Meters {
+		Meters::new(self.0.meter(), self.1.meter())
+	}
+}
+
+impl<T: Send + Sync + 'static + Debug> overseer::SubsystemSender<T> for ToWorker<T> {
+	fn send_message<'life0, 'async_trait>(
+		&'life0 mut self,
+		msg: T,
+	) -> ::core::pin::Pin<
+		Box<dyn ::core::future::Future<Output = ()> + ::core::marker::Send + 'async_trait>,
+	>
+	where
+		'life0: 'async_trait,
+		Self: 'async_trait,
+	{
+		async {
+			if let Err(err) =
+				self.0.send(polkadot_overseer::FromOrchestra::Communication { msg }).await
+			{
+				gum::error!(
+					target: LOG_TARGET,
+					"Failed to send message to approval voting worker: {:?}, subsystem is probably shutting down.",
+					err
+				);
+			}
+		}
+		.boxed()
+	}
+
+	fn try_send_message(&mut self, msg: T) -> Result<(), metered::TrySendError<T>> {
+		self.0
+			.try_send(polkadot_overseer::FromOrchestra::Communication { msg })
+			.map_err(|result| {
+				let is_full = result.is_full();
+				let msg = match result.into_inner() {
+					polkadot_overseer::FromOrchestra::Signal(_) =>
+						panic!("Cannot happen variant is never built"),
+					polkadot_overseer::FromOrchestra::Communication { msg } => msg,
+				};
+				if is_full {
+					metered::TrySendError::Full(msg)
+				} else {
+					metered::TrySendError::Closed(msg)
+				}
+			})
+	}
+
+	fn send_messages<'life0, 'async_trait, I>(
+		&'life0 mut self,
+		msgs: I,
+	) -> ::core::pin::Pin<
+		Box<dyn ::core::future::Future<Output = ()> + ::core::marker::Send + 'async_trait>,
+	>
+	where
+		I: IntoIterator<Item = T> + Send,
+		I::IntoIter: Send,
+		I: 'async_trait,
+		'life0: 'async_trait,
+		Self: 'async_trait,
+	{
+		async {
+			for msg in msgs {
+				self.send_message(msg).await;
+			}
+		}
+		.boxed()
+	}
+
+	fn send_unbounded_message(&mut self, msg: T) {
+		if let Err(err) =
+			self.1.unbounded_send(polkadot_overseer::FromOrchestra::Communication { msg })
+		{
+			gum::error!(
+				target: LOG_TARGET,
+				"Failed to send unbounded message to approval voting worker: {:?}, subsystem is probably shutting down.",
+				err
+			);
+		}
+	}
+
+	fn send_message_with_priority<'life0, 'async_trait, P>(
+		&'life0 mut self,
+		msg: T,
+	) -> ::core::pin::Pin<
+		Box<dyn ::core::future::Future<Output = ()> + ::core::marker::Send + 'async_trait>,
+	>
+	where
+		P: 'async_trait + Priority,
+		'life0: 'async_trait,
+		Self: 'async_trait,
+	{
+		match P::priority() {
+			polkadot_overseer::PriorityLevel::Normal => self.send_message(msg),
+			polkadot_overseer::PriorityLevel::High =>
+				async { self.send_unbounded_message(msg) }.boxed(),
+		}
+	}
+
+	fn try_send_message_with_priority<P: Priority>(
+		&mut self,
+		msg: T,
+	) -> Result<(), metered::TrySendError<T>> {
+		match P::priority() {
+			polkadot_overseer::PriorityLevel::Normal => self.try_send_message(msg),
+			polkadot_overseer::PriorityLevel::High => Ok(self.send_unbounded_message(msg)),
```

Comment:
This is implementing the subsystem trait for ToWorker, which does not have priority channels, because they are not needed. They could be added in the future if they are truly needed, but till then I opted for the minimal implementation.